csharp/src/Microsoft.ML.OnnxRuntime/ProviderOptions.cs

```diff
@@ -0,0 +1,143 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.InteropServices;
+
+namespace Microsoft.ML.OnnxRuntime
+{
+    /// <summary>
+    /// Search Algorithm for Cudnn Conv.
+    /// </summary>
+    public enum OrtCudnnConvAlgoSearch
+    {
+        EXHAUSTIVE,  //!< expensive exhaustive benchmarking using cudnnFindConvolutionForwardAlgorithmEx
+        HEURISTIC,   //!< lightweight heuristic based search using cudnnGetConvolutionForwardAlgorithm_v7
+        DEFAULT,     //!< default algorithm using CUDNN_CONVOLUTION_FWD_ALGO_IMPLICIT_PRECOMP_GEMM
+    }
+
+    /// <summary>
+    /// Provider options for CUDA EP.
+    /// </summary>
+    public struct OrtCUDAProviderOptions
```

Comment:
>OrtCUDAProviderOptions [](start = 18, length = 22)

This seems to be a duplicate class. Have you considered to use the same class everywhere?

Reply:
Made only one OrtCUDAProviderOptions exist.